lib/storage/metadata/mongoclient/MongoClientInterface.ts

```
@@ -859,45 +871,71 @@ class MongoClientInterface {
             ordered: true,
         })
             .then(() => cb(null, `{"versionId": "${versionId}"}`))
-            .catch((err) => {
+            .catch((err: MongoBulkWriteError) => {
                 /*
                  * Related to https://jira.mongodb.org/browse/SERVER-14322
                  * It happens when we are pushing two versions "at the same time"
                  * and the master one does not exist. In MongoDB, two threads are
                  * trying to create the same key, the master version, and one of
                  * them, the one with the highest versionID (less recent one),
                  * fails.
-                 * We check here than than the MongoDB error is related to the
-                 * second operation, the master version update and than the error
-                 * code is the one related to mentionned issue.
```

this comment looks somewhat good (except for typos)?
I think it would be good to keep such very high-level overview (i.e. we ignore error on master document, if the version document was successfully written)

---

I adapted it a bit, because it was really discussing a mongodb defect already fixed. The new version will present the risk of version ID conflict and the two cases: we can fail either at the first, or the second operation, for different reasons.